jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -87,4 +97,53 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
             }
         )
     }
+
+    fun checkRequiredPermissions(project: Project, clusterArn: String, taskArn: String) {
```

Comment:
this method is super dense

please break it up into smaller methods and add comments

Reply:
okay, will do that!